2020-08-17 21:19:03,708 [1] ?     - Application - Main is invoked

2020-08-17 21:19:12,300 [5] ?     - error! winter is coming!

2020-08-17 21:19:12,301 [5] ?     - critical! winter is coming!

2020-08-17 21:30:46,480 [1] ?     - Application - Main is invoked

2020-08-17 21:33:02,613 [1] ?     - Application - Main is invoked

2020-08-17 21:33:16,451 [9] ?     - error! winter is coming!

2020-08-17 21:33:16,452 [9] ?     - critical! winter is coming!

2020-08-17 21:33:44,123 [8] ?     - error! winter is coming!

2020-08-17 21:33:44,124 [8] ?     - critical! winter is coming!

2020-08-17 21:37:37,184 [1] ?     - Application - Main is invoked

2020-08-17 21:37:46,222 [4] ?     - error! winter is coming!

2020-08-17 21:37:46,224 [4] ?     - critical! winter is coming!

2020-08-17 21:37:53,965 [22] ?     - error! winter is coming!

2020-08-17 21:37:53,966 [22] ?     - critical! winter is coming!

2020-08-17 21:58:02,417 [1] ?     - Application - Main is invoked

2020-08-17 21:59:24,971 [1] ?     - Application - Main is invoked

2020-08-17 22:00:54,384 [1] ?     - Application - Main is invoked

2020-08-17 22:01:00,074 [4] ?     - error! winter is coming!

2020-08-17 22:01:00,077 [4] ?     - critical! winter is coming!

2020-08-17 22:11:17,207 [1] ?     - Application - Main is invoked

2020-08-17 22:11:17,926 [4] ?     - error! winter is coming!

2020-08-17 22:11:17,927 [4] ?     - critical! winter is coming!

2020-08-17 22:18:04,930 [1] ?     - Application - Main is invoked

2020-08-17 22:18:10,031 [14] ?     - error! winter is coming!

2020-08-17 22:18:10,033 [14] ?     - critical! winter is coming!

2020-08-17 22:21:32,371 [1] ?     - Application - Main is invoked

2020-08-17 22:21:33,047 [10] ?     - error! winter is coming!

2020-08-17 22:21:33,048 [10] ?     - critical! winter is coming!

2020-08-17 22:21:47,868 [9] ?     - error! winter is coming!

2020-08-17 22:21:47,869 [9] ?     - critical! winter is coming!

2020-08-17 22:44:47,343 [1] ?     - Application - Main is invoked

2020-08-17 22:44:52,899 [17] ?     - error! winter is coming!

2020-08-17 22:44:52,900 [17] ?     - critical! winter is coming!

2020-08-17 22:44:53,281 [10] ?     - The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml

System.InvalidOperationException: The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml
   at Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable`1 originalLocations)
   at Microsoft.AspNetCore.Mvc.ViewComponents.ViewViewComponentResult.ExecuteAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentInvoker.InvokeAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentHelper.InvokeCoreAsync(ViewComponentDescriptor descriptor, Object arguments)
   at AspNetCore.Views_Shared__LayoutBreadcrumbs.ExecuteAsync() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Views\Shared\_LayoutBreadcrumbs.cshtml:line 48
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebStore.Startup.<>c.<<UseSampleErrorCheck>b__4_0>d.MoveNext() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Startup.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at WebStore.Infrastructure.TokenMiddleware.InvokeAsync(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Infrastructure\TokenMiddleware.cs:line 25
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatcher|8_0(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task`1 matcherTask)
   at WebStore.Services.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\Services\WebStore.Services\ErrorHandlingMiddleware.cs:line 24
2020-08-17 22:44:56,131 [9] ?     - The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml

System.InvalidOperationException: The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml
   at Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable`1 originalLocations)
   at Microsoft.AspNetCore.Mvc.ViewComponents.ViewViewComponentResult.ExecuteAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentInvoker.InvokeAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentHelper.InvokeCoreAsync(ViewComponentDescriptor descriptor, Object arguments)
   at AspNetCore.Views_Shared__LayoutBreadcrumbs.ExecuteAsync() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Views\Shared\_LayoutBreadcrumbs.cshtml:line 48
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebStore.Startup.<>c.<<UseSampleErrorCheck>b__4_0>d.MoveNext() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Startup.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at WebStore.Infrastructure.TokenMiddleware.InvokeAsync(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Infrastructure\TokenMiddleware.cs:line 25
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at WebStore.Services.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\Services\WebStore.Services\ErrorHandlingMiddleware.cs:line 24
2020-08-17 22:44:56,135 [9] ?     - An unhandled exception has occurred while executing the request.

2020-08-17 22:46:46,762 [1] ?     - Application - Main is invoked

2020-08-17 22:47:01,035 [14] ?     - Object reference not set to an instance of an object.

System.NullReferenceException: Object reference not set to an instance of an object.
   at AspNetCore.Views_Shared__LayoutBreadcrumbs.ExecuteAsync() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Views\Shared\_LayoutBreadcrumbs.cshtml:line 41
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebStore.Startup.<>c.<<UseSampleErrorCheck>b__4_0>d.MoveNext() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Startup.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at WebStore.Infrastructure.TokenMiddleware.InvokeAsync(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Infrastructure\TokenMiddleware.cs:line 25
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at WebStore.Services.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\Services\WebStore.Services\ErrorHandlingMiddleware.cs:line 24
2020-08-17 22:47:01,090 [14] ?     - An unhandled exception has occurred while executing the request.

2020-08-17 22:47:35,606 [1] ?     - Application - Main is invoked

2020-08-17 22:47:36,715 [9] ?     - The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml

System.InvalidOperationException: The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml
   at Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable`1 originalLocations)
   at Microsoft.AspNetCore.Mvc.ViewComponents.ViewViewComponentResult.ExecuteAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentInvoker.InvokeAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentHelper.InvokeCoreAsync(ViewComponentDescriptor descriptor, Object arguments)
   at AspNetCore.Views_Shared__LayoutBreadcrumbs.ExecuteAsync() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Views\Shared\_LayoutBreadcrumbs.cshtml:line 48
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebStore.Startup.<>c.<<UseSampleErrorCheck>b__4_0>d.MoveNext() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Startup.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at WebStore.Infrastructure.TokenMiddleware.InvokeAsync(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Infrastructure\TokenMiddleware.cs:line 25
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at WebStore.Services.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\Services\WebStore.Services\ErrorHandlingMiddleware.cs:line 24
2020-08-17 22:47:36,764 [9] ?     - An unhandled exception has occurred while executing the request.

2020-08-17 22:48:30,472 [1] ?     - Application - Main is invoked

2020-08-17 22:48:31,587 [4] ?     - The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml

System.InvalidOperationException: The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml
   at Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable`1 originalLocations)
   at Microsoft.AspNetCore.Mvc.ViewComponents.ViewViewComponentResult.ExecuteAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentInvoker.InvokeAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentHelper.InvokeCoreAsync(ViewComponentDescriptor descriptor, Object arguments)
   at AspNetCore.Views_Shared__LayoutBreadcrumbs.ExecuteAsync() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Views\Shared\_LayoutBreadcrumbs.cshtml:line 47
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebStore.Startup.<>c.<<UseSampleErrorCheck>b__4_0>d.MoveNext() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Startup.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at WebStore.Infrastructure.TokenMiddleware.InvokeAsync(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Infrastructure\TokenMiddleware.cs:line 25
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at WebStore.Services.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\Services\WebStore.Services\ErrorHandlingMiddleware.cs:line 24
2020-08-17 22:48:31,671 [4] ?     - An unhandled exception has occurred while executing the request.

2020-08-17 22:49:03,183 [1] ?     - Application - Main is invoked

2020-08-17 22:49:07,794 [5] ?     - error! winter is coming!

2020-08-17 22:49:07,797 [5] ?     - critical! winter is coming!

2020-08-17 22:50:38,027 [4] ?     - The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml

System.InvalidOperationException: The view 'Components/BreadCrumbs/Default' was not found. The following locations were searched:
/Views/Catalog/Components/BreadCrumbs/Default.cshtml
/Views/Shared/Components/BreadCrumbs/Default.cshtml
/Pages/Shared/Components/BreadCrumbs/Default.cshtml
   at Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable`1 originalLocations)
   at Microsoft.AspNetCore.Mvc.ViewComponents.ViewViewComponentResult.ExecuteAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentInvoker.InvokeAsync(ViewComponentContext context)
   at Microsoft.AspNetCore.Mvc.ViewComponents.DefaultViewComponentHelper.InvokeCoreAsync(ViewComponentDescriptor descriptor, Object arguments)
   at AspNetCore.Views_Shared__LayoutBreadcrumbs.ExecuteAsync() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Views\Shared\_LayoutBreadcrumbs.cshtml:line 47
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebStore.Startup.<>c.<<UseSampleErrorCheck>b__4_0>d.MoveNext() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Startup.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at WebStore.Infrastructure.TokenMiddleware.InvokeAsync(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Infrastructure\TokenMiddleware.cs:line 25
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at WebStore.Services.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\Services\WebStore.Services\ErrorHandlingMiddleware.cs:line 24
2020-08-17 22:50:38,158 [4] ?     - An unhandled exception has occurred while executing the request.

2020-08-17 22:53:38,456 [1] ?     - Application - Main is invoked

2020-08-17 22:53:39,623 [12] ?     - error! winter is coming!

2020-08-17 22:53:39,626 [12] ?     - critical! winter is coming!

2020-08-17 22:54:10,857 [12] ?     - Object reference not set to an instance of an object.

System.NullReferenceException: Object reference not set to an instance of an object.
   at AspNetCore.Views_Shared__LayoutBreadcrumbs.ExecuteAsync() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Views\Shared\_LayoutBreadcrumbs.cshtml:line 41
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(IRazorPage page, ViewContext context, Boolean invokeViewStarts)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderLayoutAsync(ViewContext context, ViewBufferTextWriter bodyWriter)
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderAsync(ViewContext context)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ViewContext viewContext, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewExecutor.ExecuteAsync(ActionContext actionContext, IView view, ViewDataDictionary viewData, ITempDataDictionary tempData, String contentType, Nullable`1 statusCode)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebStore.Startup.<>c.<<UseSampleErrorCheck>b__4_0>d.MoveNext() in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Startup.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
   at WebStore.Infrastructure.TokenMiddleware.InvokeAsync(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\UI\WebStore_2020\Infrastructure\TokenMiddleware.cs:line 25
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at WebStore.Services.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\Users\kiruh\source\repos\WebStore_2020\Services\WebStore.Services\ErrorHandlingMiddleware.cs:line 24
2020-08-17 22:54:10,900 [12] ?     - An unhandled exception has occurred while executing the request.

2020-08-17 22:54:39,479 [24] ?     - error! winter is coming!

2020-08-17 22:54:39,480 [24] ?     - critical! winter is coming!

